docs/guides/12-Vue-integration.md

```diff
@@ -0,0 +1,197 @@
+
+```
+npm i @carto/airship-style
+```
+
+To import the styles, we need to include `airship.css` in the file:
+```js
+import '@carto/airship-style/dist/airship.css';
+```
+
+## Integrating Web Components
+First, we need to tell Vue to ignore elements starting with `as-`, which are our Airship components. We don't want Vue to raise an error everytime an Airship component is included in a template.
+
+```js
+import Vue from 'vue';
+
+Vue.config.ignoredElements = [/as-\w+/];
+```
+
+Second, we need to import and define our Airship components. As we did before with Airship styles, we import them from npm, and then we need to execute `defineCustomElements` function.
+
+```js
+import { defineCustomElements } from '@carto/airship-components';
+defineCustomElements(window);
+```
+
+### Using Web Components in templates
+To add a Web Component to a Vue template you just need to use the HTML syntax that is provided in the reference.
+
+Although Web Components are supported in Vue, there's one caveat when it comes to injecting data into the component. The only properties that can be passed via attributes are text and number properties, whether via direct text or injecting the property in the template.
+
+```html
+<as-category-widget
+  heading="Business Volume"
+  description={widgetDescription}>
+</as-category-widget>
+```
+
+To pass complex properties such as Objects and Arrays or listen to component events, you need to do it via Vue element reference or creating a mirror component.
+
+#### Managing component with element reference
+There are some properties that we cannot pass or events that we cannot listen to in the example above, such as category property or categoriesSelected event.
+
+To get the element reference, we need to set `ref` property in the component template node.
+
+```html
+<template>
+  <as-category-widget
+    ref="categoryWidget"
+    heading="Business Volume"
+    description={widgetDescription}>
+  </as-category-widget>
+</template>
+```
+
+Element's reference will be in our Vue component's `refs` property, hence we will have full access to our Airship component.
+
+##### Listening to component events
+The best way to listen to component events is in `mounted` function in your desired component.
+
+```js
+export default {
+  name: 'MyComponent',
+  mounted: function () {
+    this.categoryWidget = this.$refs.categoryWidget;
+
+    this.categoryWidget.addEventListener('categoriesSelected', (event) => {
+      console.log('Selected Categories', event.detail);
+    });
+  }
+}
+```
+
+We create an easy-to-use reference to the widget so that we can reuse it in other parts of our component. Then, we listen to `categoriesSelected` event where you can put your own custom code to perform the action that you want.
+
+It will only listen once to component event and it won't create multiple unneeded event listeners by doing it in `mounted`callback`.
```

Comment:
Any mention on where to remove the listener? Or if Vue takes care of that automagically, mention that the developer should not worry about it.

Reply:
Any framework should take care of it when destroying the component where the Web Component was placed